frontend/src/js/previous-queries/upload/UploadQueryResults.tsx

```diff
@@ -59,18 +58,20 @@ const UploadQueryResults = ({
       setLoading(true);
 
       const result = await postQueryUpload(datasetId, query);
-
       setUploadResult(result);
 
       loadQueries(datasetId);
     } catch (e) {
-      if ((e as { status?: number }).status === 400) {
+      if (
+        (e as { status?: number }).status === 400 &&
+        "resolved" in (e as object)
+      ) {
         setUploadResult(e as UploadQueryResponseT);
       } else {
         dispatch(
           setMessage({
             message: t("uploadQueryResultsModal.uploadFailed"),
-            type: SnackMessageType.ERROR,
+            type: "error",
           }),
         );
       }
```

Comment:
@awildturtok so if I understand this correctly, we seem to allow 400s as valid responses. But I received an internal server error that was still using a 400 status code, not a 500. So I've added a `"resolved" in e` check, which is far from ideal. Hmm.

<img width="458" alt="Screenshot 2024-05-14 at 11 53 46" src="https://github.com/ingef/conquery/assets/1403093/829d1990-e15f-4a21-9d67-6852b8ae396f">

Reply:
Do I understand it correctly, that you would expect a 200ish response if nothing could be uploaded?

Reply:
Yes, together with the following response shape
```
export type UploadQueryResponseT = {
  resolved: number;
  unresolvedId: string[][];
  unreadableDate: string[][];
};
```

and in case it's an internal server error, I'd expect a 500, not a 4xx.